pandas/core/internals/blocks.py

```diff
@@ -3005,6 +3005,30 @@ def where(
             )
         return result
 
+    def replace(
+        self, to_replace, value, inplace=False, filter=None, regex=False, convert=True
+    ):
+        result = self if inplace else self.copy()
+        if filter is None:
+            categories = result.values.categories.tolist()
+            if to_replace in categories:
```

Comment:
can we not define a Categorical.replace to handle this code here (from 3013 thru 3020)

Reply:
done, with tests added in `pandas/tests/arrays/categorical/test_algos.py`